pkg/agent/agent_windows.go

```diff
@@ -32,3 +40,99 @@ func (i *Initializer) setupExternalConnectivity() error {
 	}
 	return nil
 }
+
+// prepareHostNetworking creates HNS Network for containers.
+func (i *Initializer) prepareHostNetworking() error {
+	return util.PrepareHostNetworking(i.nodeConfig.PodCIDR, i.nodeConfig.NodeIPAddr)
+}
+
+// prepareOVSBridge adds local port and uplink to ovs bridge.
+func (i *Initializer) prepareOVSBridge() error {
+	hnsNetwork, err := hcsshim.GetHNSNetworkByName(util.LocalHNSNetwork)
+	if err != nil {
+		return err
+	}
+
+	// Set datapathID of OVS bridge.
+	// If no datapathID configured explicitly, the reconfiguration operation will change OVS bridge datapathID
+	// and break the OpenFlow channel.
+	datapathID := strings.Replace(hnsNetwork.SourceMac, ":", "", -1)
+	datapathID = "0000" + datapathID
+	if err = i.ovsBridgeClient.SetDatapathID(datapathID); err != nil {
+		klog.Errorf("Failed to set datapath_id %s: %v", datapathID, err)
+		return err
+	}
+
+	// Create local port.
+	brName := i.ovsBridgeClient.GetBridgeName()
+	if _, err := i.ovsBridgeClient.GetOFPort(brName); err == nil {
+		klog.Infof("OVS bridge local port %s already exists, skip the configuration", brName)
+	} else {
+		if _, err := i.ovsBridgeClient.CreateInternalPort(brName, config.AutoAssignedOFPort, nil); err != nil {
```

Comment:
Question - why not use a fixed ofport, which is simpler?

Reply:
Creating local port does not receive "ofport_request" param, so we use config.AutoAssignedOFPort=0 to ignore this param.

Reply:
OK.

Reply:
Better to add some comments for this.

Reply:
Would you add comments here to explain why AutoAssignedOFPort is needed?